### DISCUSSION TITLE
NoReferencedTableError: Foreign key associated with column … could not find table …

### DISCUSSION THREAD

**Discussion by Shandeika** [sqlalchemy/sqlalchemy#8786] [Usage Questions] [answered]
At startup I get:
```
sqlalchemy.exc.NoReferencedTableError: Foreign key associated with column 'channels.category' could not find table 'categories' with which to generate a foreign key to target column 'id'
```
In my understanding, this table is created before the other one, because of this an error occurs, but this is only an assumption.
A repository with reproducible code. An empty database is used, all data must be entered in config.py
https://github.com/Shandeika/testSQLA

**Comment by Shandeika** [comment]
It just so happened that I'm a little stupid and put the password from the database into the git :facepalm:
Since it so happened, you can test on my database

**Comment by CaselIT** [comment, marked as answer]
Categories is not imported in the init. https://github.com/Shandeika/testSQLA/blob/master/database/models/__init__.py
if no-one imports the module then python will not execute it and the category table will not be found because it was never imported

**Reply by Shandeika** [reply]
I think I've already blunted it 2 times. And how I didn't notice it before...
sorry for the time spent
